dojo/tools/asff/parser.py

```diff
@@ -41,9 +41,35 @@ def get_findings(self, file, test):
             else:
                 active = False
 
+            # Adding the Resources:0/Id value to the description.
+            #
+            # This is needed because every Finding in AWS from Security Hub has an
+            # associated ResourceId that contains the full AWS ARN and without it,
+            # it is much more difficult to track down the specific resource.
+            #
+            # This is different from the Finding Id - as that is from the Security Hub
+            # control and has no information about the offending resource.
+            #
+            # Retrieve the AWS ARN / Resource Id
+            resource_arns = []
+            resource_ids = item.get("Resources", [])
+            for resource in resource_ids:
+                resource_id = resource.get("Id")
+                if resource_id:
+                    resource_arns.append(resource_id)
```

Comment:
To fix the unit tests, you'll want to update the line [here](https://github.com/DefectDojo/django-DefectDojo/blob/b25c66676d4f0a88dff66911960ebb78bbf26715/unittests/tools/test_asff_parser.py#L21) to use the new description format. You could refactor the parser changes a bit, using a helper method to retrieve the ARNs from a given `item` and leveraging that in the unit test along with the format string that actually populates it in the description so you don't have to copy the whole implementation over. Roughly something like:

```python
def get_item_resource_arns(self, item):
    ... your current code for walking the item's resources to collect all the ARNs on line 54-59 ...
    return resource_arns
```

And then reference that here:

```suggestion
            resource_arns = self.get_item_resource_arns(item)
```

Then do the same in the unit test, collecting the ARNs and referencing them in a new description format string to match the parser's formatting. You'll need to pass `parser` into `common_check_finding` in the calls from the [`test_asff_one_vuln`](https://github.com/DefectDojo/django-DefectDojo/blob/b25c66676d4f0a88dff66911960ebb78bbf26715/unittests/tools/test_asff_parser.py#L43), [`test_asff_many_vulns`](https://github.com/DefectDojo/django-DefectDojo/blob/b25c66676d4f0a88dff66911960ebb78bbf26715/unittests/tools/test_asff_parser.py#L52), and [`test_asff_guardduty`](https://github.com/DefectDojo/django-DefectDojo/blob/b25c66676d4f0a88dff66911960ebb78bbf26715/unittests/tools/test_asff_parser.py#L61) methods so you can reference `parser.get_item_resource_arns` from `common_check_finding`.

Reply:
I believe I have refactored the parser as you have suggested (thank you again!).  I also passed in the parser = AsffParser() on line 20 in the common_check_finding.

I'm now in the process of going through those load_sample_json files and collecting the ARNs so I can reference them in the new description format string.

Would this change take place on L22 and editing:
self.assertEqual(finding.description, data[index]["Description"])

or do I need to capture each respective load_sample_json ARN + description and within those methods pass a self.assertEqual value for the finding.description with the new value format of:
"**AWS resource ARN:** {resource_arn_strings}\n\n{control_description}"